vllm/v1/attention/backends/flashinfer.py

```diff
@@ -569,14 +578,62 @@ def build(self,
             use_cascade=use_cascade,
             shared_qo_indptr_cpu=shared_qo_indptr_cpu,
             shared_kv_page_indptr_cpu=shared_kv_page_indptr_cpu,
-            shared_kv_page_indices_cpu=shared_kv_page_indices_cpu,
+            shared_kv_page_indices=shared_kv_page_indices,
             shared_kv_last_page_len_cpu=shared_kv_last_page_len_cpu,
+            _update_paged_kv_indices_=_update_paged_kv_indices_,
         )
 
         self._plan(attn_metadata)
 
         return attn_metadata
 
+    def update_block_table(self, metadata: FlashInferMetadata,
+                           blk_table: torch.Tensor,
+                           slot_mapping: torch.Tensor):
+        new = copy.copy(metadata)
+        # We only made a shallow copy of the metadata, so we need to allocate
+        # a new paged_kv_indices tensor.
+        new.paged_kv_indices = torch.empty_like(metadata.paged_kv_indices)
```

Comment:
Allocations can move to class constructor based on maximum paged_kv_indices and here we can do a tensor slice to the required shape